test/test_factory.py

```diff
@@ -172,6 +173,27 @@ def _(loc, scale, value):
     check_funsor(actual, {"i": Bint[3]}, Real)
 
 
+@pytest.mark.skipif(get_backend() != "torch", reason="requires nn.Module")
+def test_nn_module():
+    import torch
+
+    class Matmul(torch.nn.Module):
+        def forward(
+            self,
+            x: Funsor,
```

Comment:
Is this a pattern (Funsor terms written as `nn.Module`s) we want to support? An alternative would be to raise an error when passing `Module` instances to `make_funsor` and request that they be converted to `Op`s instead.

Reply:
I don't see any reason to not support `Funsor` subclasses based on `nn.Module` s.  Forcing users to make an op would force them into the world of positional tensors.  But I haven't thought this through entirely.